db.go

```diff
@@ -69,6 +69,14 @@ func (db *DB) Insert(collectionName string, docs ...*Document) error {
 	return db.engine.Insert(collectionName, insertDocs...)
 }
 
+// Save or update a document
+func (db *DB) Save(collectionName string, doc *Document) error {
+	if doc.Get(objectIdField) == "" {
+		doc.Set(objectIdField, newObjectId())
```

Comment:
In this case, simply call the db.Insert(). You don't even need to set the objectId yourself.